### PR TITLE
py-netcdf4: a couple of improvements

### PR DESCRIPTION
**Pull request by skosukhin**
- avoid recompilation
- make the older versions to work with `hdf5@1.10:`

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-netcdf4/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-netcdf4/package.py:43]: I've noticed this as well, I'm thinking about switching the default `phases` to just install, but want to wait until we have better CI to try something like that.

**Review by adamjstewart — APPROVED** [review]
Would you have any interest in being an official "maintainer" for this package?

**Comment by skosukhin** [comment]
@adamjstewart I also took the liberty to add the latest version.

**Review by adamjstewart — APPROVED** [review]

**Comment by skosukhin** [comment]
BTW, once/if #8623 is merged, we can replace `patch('disable_pkgconf.patch')` with `depends_on('pkgconf', type='build')`.

**Pull request merged**
